sellercenter-attributes/src/main/java/com/gfg/sellercenter/attributes/reader/HttpReader.java

```diff
@@ -0,0 +1,56 @@
+package com.gfg.sellercenter.attributes.reader;
+
+import lombok.AllArgsConstructor;
+
+import org.json.JSONArray;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class HttpReader {
+
+    private final String fqdnUrl;
+
+    /**
+     * This json array represent a list of products and attributes information
+     *
+     * @param productIds a list of product ids in order to fetch related attribute which has options which requires a serial number
+     * @return JSONArray It retrieves a json array with a product and attributes information
+     * @throws Exception throws an exception when the endpoint is not reachable
+     */
+    public JSONArray getSerialNumberRequiredAttributesByProductIds(List<Integer> productIds)
+            throws Exception {
+        String productIdsConverted =
```

Comment:
`(new JSONArray(productIds)).toString()` will do the thing. At the end this request parameter should be json string and what can create better json string than native serializer.

Reply:
Also, just as a precaution: can you please add filtering non-unique values?

Reply:
The uniqueness is done in the service already.

Reply:
Yes and no :) When we started to test java service on staging with Waikiki seller (it has dozens of requests with tens of thousands of products) we faced an issue on every step: too long GET string to send, too long GET string to parse, too much parameters in SQL etc. If some uncautious developer will try to call this method with unfiltered list of products request will not even reach Graph service because of its length.

Reply:
No what i meant was from my Service class ^^

Reply:
I understand that, therefore it is just "precautions". Normally you won't call library with `getSerialNumberRequiredAttributesByProductIds(123, 123, 123, 123)`. If you more like to forgive mistakes of users of that library - please add filtering, if you think otherwise - leave it like it is. Both variants are ok to me.